index.html

```diff
@@ -449,6 +449,22 @@ <h2>Algorithms</h2>
 that utilize the Elliptic Curve Digital Signature Algorithm (ECDSA) [[FIPS-186-5]].
       </p>
 
+      <p>
+Implementations SHOULD fetch and cache <a>verification method</a> information as
+early as possible when adding or verifying proofs. Parameters passed to
+functions in this section utilize information from the <a>verification
+method</a>, such as the public key size, to determine function parameters, such
+as the cryptographic hashing algorithm to utilize.
+      </p>
+
+      <p>
+When the RDF Dataset Canonicalization Algorithm [[RDF-CANON]] is used with ECDSA
+algorithms, the cryptographic hashing function that is passed to the algorithm
+MUST be determined by the size of the associated public key. For P-256 keys,
+SHA-2 with 256 bits of output is utilized. For P-384 keys, SHA-2 with 384-bits
+of output is utilized.
```

Comment:
I have a problem with this.

Looking [RDFC-1.0](https://w3c.github.io/rdf-canon/spec/#dfn-hash-algorithm) what RDF-CANON says is that SHA-256 _is_ the hash used inside, and, essentially, that the implementation _MAY_ have a parametrization of that algorithm. It does not even say _which_ additional algorithms should be available (my implementation is limited to those that the current openssh offers, for example, I do not implement hashing myself). With this, I believe that the approach proposed here will break interoperability.

Also: why this? The hash function used within RDFC-1.0 is purely internal to canonicalization. It is not connected to the key size this section talks about. Maybe what you mean is that the hash function you discuss should be used to hash the _result_ of the RDF canonizalization, ie, to hash the canonical n-quads. But that particular step is _not_ what RDFC-1.0 does...

Reply:
Because I am not a domain expert, so maybe you really meant to influence the internals of RDFC 1.0 by instructing it to use something else than the default SH-256. But then, some slightly hand-wavy text should be added along the lines of "the environment should choose an RDFC implementation that is prepared for the usage of the particular hash function that the environment requires". Or something like that.

Reply:
I think you are right @iherman RDFC doesn't seem to take a parameter. The first paragraph seems fine since it says we need to get the key to then determine the appropriate hashing algorithm for the *hashing* step. This applies to both RDFC and JCS.

Reply:
The RDFC spec specifically states: "NOTE: Implementations can be written to parameterize the hash algorithm without any other changes."

Reply:
@Wind4Greg, @msporny quotes the right place. What is not specified in the RDFC spec is _how_ the hash algorithm can be changed, ie, what the API is for this. But that is a detail.

The point I was making is that this possibility is not _required_, hence my comment in https://github.com/w3c/vc-di-ecdsa/pull/36#discussion_r1306682157

Reply:
@iherman we discussed this on the RCH WG call last week and it was decided to strengthen the language to make it more clear how one would signal the hash algorithm is changed. We say as much as we can in this spec, the RCH spec will have to say more in order to make it more clear that this algorithm requires it. If one can't do that with their RDF-CANON library, they will fail the test suite and will not be conformant.